### PR TITLE
tguin haunted eight ball

### LINKED ISSUE #25368: after using the magic eightball only works once, using it any time after blocks ghosts from voting
hack cough

even if they shake it it still says the eightball is not currently being shaken so ghosts can't vote

### LINKED ISSUE #41718: Magic 8 ball item requires ghost proximity to be able to answer
## Round ID: 
Terry 97766

## Testmerges: 
#41099

## Reproduction:
1. Chilled as a ghost. 
2. Saw "Kennard Beach is shaking the magic eightball, hoping to get an answer to "Congratulations, you made it to Orion!" (Click to help) "
3. Clicked . I got a popup with every option greyed out. So I quickly clicked "orbit" under ooc to find his name and get close. The options stopped being greyed out and I could answer. The answer was repeated IC by the ball. 

I expect ghosts to be able to answer the ball without needing proximity. This was the toy version, and I assume the traitor item is broken as well.

### PR DESCRIPTION
**Pull request by actioninja**
## About The Pull Request

Redoes the haunted eight ball ui for tgui next and gives it a pretty significant rework.
Instead of just dumping every option, ghosts vote yes no or maybe:
image
Also you shake it for 30 seconds now, so the vote time is 30 seconds.
ALSO IT ACTUALLY FUCKING FUNCTIONS NOW (probably)

closes #25368
closes #41718

## Changelog
:cl:
add: tgui-next magic eight ball
tweak: haunted eight ball is now shook for 30 seconds, which gives it a 30 second ghost vote time instead of 15
fix: Haunted eight ball actually clears its vote list after counting votes, so it functions after over two years.
/:cl:

### CODE REVIEW

File: code/game/objects/items/eightball.dm [hunk 1]
```diff
@@ -95,16 +95,48 @@
...
+			"Yes definitely",
+			"You may rely on it",
+			"As I see it, yes",
+			"Most likely",
+			"Outlook good",
+			"Yes",
+			"Signs point to yes"
+		),
+		"maybe" = list(
+			"Signs point to yes",
```
Comment (Fikou) [inline code/game/objects/items/eightball.dm]: this is in both yes and maybe

File: code/game/objects/items/eightball.dm [hunk 2]
```diff
@@ -137,38 +169,31 @@
...
+	var/top_amount = 0
+	var/top_vote
 
+	for(var/vote in votes)
+		var/amount_of_votes = length(votes[vote])
+		if(amount_of_votes > top_amount)
+			top_vote = vote
+			top_amount = amount_of_votes
+		//If one option actually has votes and there's a tie, pick between them 50/50
+		else if(top_amount && amount_of_votes == top_amount && prob(50))
```
Comment (kriskog) [inline code/game/objects/items/eightball.dm]: In a 3 way tie, the "no" option will have twice the chance of being picked due to only having to win the prob once. Its not super important.

**Comment by ATH1909** [comment]
antag freeze smh

/joke

**Comment by kriskog** [comment]
There is a 1/100 every 8-ball that spawns is haunted. Arguably higher than any antag picking this piece of shit :nerd_face:

File: code/game/objects/items/eightball.dm (hunk 2 above)
Comment (actioninja) [reply code/game/objects/items/eightball.dm]: oh shidd, I doubt a three way tie will ever happen

Comment (kriskog) [reply code/game/objects/items/eightball.dm]: It will, but its whatever. I cant think of any solution that isnt overengineered for a haunted eightball.

**Pull request merged**